archive/100-theorems-list/inverse-triangle-sum.lean

```diff
@@ -0,0 +1,87 @@
+import tactic
+import algebra.big_operators
+import data.real.basic
+
+lemma range_insert (n : ℕ) :
+insert n (finset.range n) = finset.range (n + 1) :=
+begin
+ext, rw finset.mem_range,
+simp only [finset.mem_insert, finset.mem_range],
+split; omega,
+end
+
+
+lemma discrete_FTC
```

Comment:
This is true for an `add_comm_group`, and can be easily modified for an `add_comm_monoid`. Should go to `algebra/big_operators`.

Reply:
It looks very similar to `sum_range_sub_of_monotone` but I don't see the version for regular subtraction.

Reply:
I think that we need something like
```lean
lemma sum_range_induction {M : Type*} [add_comm_monoid M] (f s : ℕ → M) (h0 : s 0 = 0) (h : ∀ n, s (n + 1) = s n + f n) (n : ℕ) : ∑ k in range n, f k = s n := sorry
```

Reply:
This lemma can be used for `sum_range_id` as well.

Reply:
I don't see anythign when i search the mathlib docs for sum_range_sub_of_monotone

Reply:
It's [here](https://leanprover-community.github.io/mathlib_docs/algebra/big_operators.html#finset.sum_range_sub_of_monotone). The google search is unfortunately often out of date.

Reply:
okay, I impliemented Yury's suggestion `sum_range_induction` and put it into algebra.big_operators. I think this is strictly stronger than sum_range_sub_monotone?